src/bin/pg_autoctl/cli_enable_disable.c

```diff
@@ -402,3 +432,593 @@ cli_disable_maintenance(int argc, char **argv)
+
+			/* display a nice summary to our users */
+			log_info("Successfully enabled new SSL configuration:");
+			log_info("  SSL is now %s",
+					 pgSetup->ssl.active ? "active" : "disabled");
+
+			if (pgSetup->ssl.createSelfSignedCert)
+			{
+				log_info("  Self-Signed certificates have been created and "
+						 "deployed in Postgres configuration settings "
+						 "ssl_key_file and ssl_cert_file");
+			}
+
+			log_warn("  Postgres connection string to the monitor "
+					 "has not been changed, see above for details");
+
+			if (updatedPrimaryConninfo)
+			{
+				log_info("  Replication connection string primary_conninfo "
+						 "has been updated to use the new sslmode \"%s\"",
+						 pgsetup_sslmode_to_string(pgSetup->ssl.sslMode));
+			}
+			break;
+		}
+
+		default:
+		{
+			log_fatal("Unrecognized configuration file \"%s\"",
+					  options.pathnames.config);
+			exit(EXIT_CODE_INTERNAL_ERROR);
+		}
+	}
+}
+
+
+/*
+ * update_ssl_configuration updates the local SSL configuration.
+ */
+static bool
+update_ssl_configuration(LocalPostgresServer *postgres, const char *nodename)
+{
+	char hbaFilePath[MAXPGPATH] = { 0 };
+	PostgresSetup *pgSetup = &(postgres->postgresSetup);
+
+	log_warn("update_ssl_configuration: ssl %s",
+			 pgSetup->ssl.active ? "on" : "off");
+
+	/*
+	 * Make sure we configure Postgres with absolute file paths.
+	 */
+	if (!normalize_filename(pgSetup->pgdata, pgSetup->pgdata, MAXPGPATH))
+	{
+		return false;
+	}
+
+	/*
+	 * When --ssl-self-signed is used, create a certificate.
+	 */
+	if (pgSetup->ssl.createSelfSignedCert &&
+		(!file_exists(pgSetup->ssl.serverKey) ||
+		 !file_exists(pgSetup->ssl.serverCert)))
+	{
+		if (!pg_create_self_signed_cert(pgSetup, nodename))
+		{
+			log_error("Failed to create SSL self-signed certificate, "
+					  "see above for details");
+			return false;
+		}
+	}
+
+	/* edit our Postgres setup to include the new SSL settings */
+	if (!postgres_add_default_settings(postgres))
+	{
+		log_error("Failed to enable SSL: "
+				  "adding default Postgres settings failed, "
+				  "see above for details");
+		return false;
+	}
+
+	/* call pg_ctl_reload() while connected as the local superuser */
+	strlcpy(pgSetup->username, "", NAMEDATALEN);
+
+	local_postgres_init(postgres, pgSetup);
+
+	if (!pgsql_reload_conf(&(postgres->sqlClient)))
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
+	/* HBA rules for hostssl are not edited */
+	sformat(hbaFilePath, MAXPGPATH, "%s/pg_hba.conf", pgSetup->pgdata);
+
+	log_warn("HBA rules in \"%s\" have NOT been edited: \"host\" "
+			 " records match either SSL or non-SSL connection attempts.",
+			 hbaFilePath);
+
+	return true;
+}
+
+
+/*
+ * update_monitor_connection_string connects to the monitor to see if ssl is
+ * active on the server. When that's the case, the function edits the monitor
+ * URI in the given KeeperConfig.
+ */
+static bool
+update_monitor_connection_string(KeeperConfig *config)
```

Comment:
This function name and comment suggest that this function updates the connection string, but actually it only logs.

Reply:
I kind of hope to fix that someday with actually doing the work, leaving in here a clue that we  would like to be able to. Maybe that warrants a better comment though...